lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,921 @@
+  contains:
+    database_name:
+      description: Database name.
+      returned: always
+      type: dict
+      sample: template0
+      contains:
+        access_priv:
+          description: Database access privileges.
+          returned: always
+          type: str
+          sample: "=c/postgres_npostgres=CTc/postgres"
+        collate:
+          description:
+          - Database collation U(https://www.postgresql.org/docs/current/collation.html).
+          returned: always
+          type: str
+          sample: en_US.UTF-8
+        ctype:
+          description:
+          - Database LC_CTYPE U(https://www.postgresql.org/docs/current/multibyte.html).
+          returned: always
+          type: str
+          sample: en_US.UTF-8
+        encoding:
+          description:
+          - Database encoding U(https://www.postgresql.org/docs/current/multibyte.html).
+          returned: always
+          type: str
+          sample: UTF8
+        owner:
+          description:
+          - Database owner U(https://www.postgresql.org/docs/current/sql-createdatabase.html).
+          returned: always
+          type: str
+          sample: postgres
+        size:
+          description: Database size in bytes.
+          returned: always
+          type: str
+          sample: 8189415
+extensions:
+  description:
+  - Extensions U(https://www.postgresql.org/docs/current/sql-createextension.html).
+  returned: always
+  type: dict
+  sample:
+  - { "plpgsql": { "description": "PL/pgSQL procedural language",
+    "extversion": { "major": 1, "minor": 0 } } }
+  contains:
+    extdescription:
+      description: Extension description.
+      returned: if existent
+      type: str
+      sample: PL/pgSQL procedural language
+    extversion:
+      description: Extension description.
+      returned: always
+      type: dict
+      contains:
+        major:
+          description: Extension major version.
+          returned: always
+          type: int
+          sample: 1
+        minor:
+          description: Extension minor version.
+          returned: always
+          type: int
+          sample: 0
+    nspname:
+      description: Namecpase where the extension is.
+      returned: always
+      type: str
+      sample: pg_catalog
+languages:
+  description: Procedural languages U(https://www.postgresql.org/docs/current/xplang.html).
+  returned: always
+  type: dict
+  sample: { "sql": { "lanacl": "", "lanowner": "postgres" } }
+  contains:
+    lanacl:
+      description:
+      - Language access privileges
+        U(https://www.postgresql.org/docs/current/catalog-pg-language.html).
+      returned: always
+      type: str
+      sample: "{postgres=UC/postgres,=U/postgres}"
+    lanowner:
+      description:
+      - Language owner U(https://www.postgresql.org/docs/current/catalog-pg-language.html).
+      returned: always
+      type: str
+      sample: postgres
+namespaces:
```

Comment:
What I find confusing, is that we have two levels of info crossing here.
Basically, this info is only for the database you are connecting to.
But it is listed on the global level, next to the databases, so it looks as if it where global.

Reply:
I would propose same fix as on extensions, but there is no pg_available_namespaces, so I would then skip this level entirely, or build it from info scanned from all databases, and adding a list of databases it was created in.

Reply:
Very good notice! I'll think about it.